src/helpers/chain.js

```diff
@@ -0,0 +1,582 @@
+  logger.info("Getting prices for " + oracle + " - " + JSON.stringify(feeds));
+
+  let fromBoard = makeFromBoard();
+  const unserializer = boardSlottingMarshaller(fromBoard.convertSlotToVal);
+  const leader = makeLeader(networkConfig.rpcAddrs[0]);
+
+  const follower = await makeFollower(`:published.wallet.${oracle}`, leader, {
+    unserializer,
+  });
+
+  let offersBalances = await getOffersAndBalances(follower, oracle);
+
+  // Get last offer id from offers from state
+  let lastOfferId = isNaN(state["last_offer_id"]) ? 0 : state["last_offer_id"];
+
+  // Initialise variable to hold results
+  let lastResults = {
+    last_offer_id: lastOfferId,
+    values: state["values"] ? state["values"] : {},
+  };
+
+  // Loop through offers starting from last visited index
+  for (let i = 0; i < offersBalances.offers.length; i++) {
+    // Get current offer
+    let currentOffer = offersBalances.offers[i];
+    let id = Number(currentOffer["status"]["id"]);
+
+    // If we found the last visited offer id in previous check, stop looping
+    if (id <= lastOfferId) {
+      break;
+    }
+
+    // If a price invitation
+    let invMakerName =
+      currentOffer["status"]["invitationSpec"]["invitationMakerName"];
+
+    if (invMakerName === "PushPrice") {
+      let feed =
+        feeds[currentOffer["status"]["invitationSpec"]["previousOffer"]];
+      let lastRound = Number(
+        currentOffer["status"]["invitationSpec"]["invitationArgs"][0]["roundId"]
+      );
+
+      // Get feeds' last observed round from state
+      let lastObservedRound = state["values"].hasOwnProperty(feed)
+        ? state["values"][feed]["round"]
+        : 0;
+
+      // If round is bigger than last observed and the offer didn't fail
+      if (
+        lastRound > lastObservedRound &&
+        !currentOffer["status"].hasOwnProperty("error")
+      ) {
+        // If id is bigger than last offer id in state, set it
+        lastResults["last_offer_id"] = id;
+        lastOfferId = id;
+
+        let price =
+          Number(
+            currentOffer["status"]["invitationSpec"]["invitationArgs"][0][
+              "unitPrice"
+            ]
+          ) / amountsIn[feed];
+
+        // Fill results variable
+        lastResults["values"][feed] = {
+          price: price,
+          id: id,
+          round: lastRound,
+        };
+        state = lastResults;
+
+        // Get latest feed price
+        let feedPrice = await queryPrice(feed);
+        // Update metrics
+        metrics.updateMetrics(
+          oracleDetails["oracleName"],
+          oracle,
+          feed,
+          price,
+          id,
+          feedPrice,
+          lastRound
+        );
+      }
+    }
+  }
+
+  // Loop through balances
+  for (let i = 0; i < offersBalances.balances.length; i++) {
+    let currentBalance = offersBalances.balances[i];
+
+    let brand = currentBalance.brand.iface.split(" ")[1];
+    if (brand.includes("BLD") || brand.includes("IST")) {
```

Comment:
Why are we only tracking balances that are agoric native only? In these scenarios I would expect a comment. Most comments like // Loop through balance are not really useful

Reply:
Fixed in https://github.com/jacquesvcritien/agoric-cl-middleware/commit/672f0f1584e452f9867209af09773164dd2bc949